--- origsrc/kazehakase-0.5.4/configure.ac	2008-03-29 03:20:42.000000000 -0500
+++ src/kazehakase-0.5.4/configure.ac	2008-06-02 22:15:00.000000000 -0500
@@ -22,13 +22,16 @@
 case "$host" in
   *-*-mingw*|*-*-cygwin*)
     platform_win32=yes
+    no_undefined="-no-undefined"
     ;;
   *)
     platform_win32=no
+    no_undefined=
     ;;
 esac
 AC_MSG_RESULT([$platform_win32])
 
+AC_SUBST(no_undefined)
 AM_CONDITIONAL(OS_WIN32, test "$os_win32" = "yes")
 AM_CONDITIONAL(PLATFORM_WIN32, test "$platform_win32" = "yes")
 
@@ -417,10 +420,15 @@
   RUBY_CFLAGS="$RUBY_CFLAGS $rbconfig_CFLAGS "
   RUBY_LIBS="$rbconfig_LIBRUBYARG $rbconfig_LIBS"
   RUBY_LDFLAGS="-L$rbconfig_libdir"
+  RUBY_EXT_LDFLAGS="-L$rbconfig_sitearchdir"
+  if test "$platform_win32" = "yes"; then
+    RUBY_LIBS="$RUBY_LIBS -lruby-gtk2 -lruby-glib2"
+  fi
 
   AC_SUBST(RUBY_CFLAGS)
   AC_SUBST(RUBY_LIBS)
   AC_SUBST(RUBY_LDFLAGS)
+  AC_SUBST(RUBY_EXT_LDFLAGS)
 
   _SAVE_CFLAGS=$CFLAGS
   ruby_undef_package_macros="
--- origsrc/kazehakase-0.5.4/etc/kz-ui-expert.xml	2008-03-29 03:21:03.000000000 -0500
+++ src/kazehakase-0.5.4/etc/kz-ui-expert.xml	2008-06-03 00:30:43.000000000 -0500
@@ -128,10 +128,10 @@
       </menu>
       <separator name="tools-sep3" />
       <menu name="ExtensionMenu" action="StockExtensionMenu">
-<!--
+
         <menuitem name="RubyDialog" action="RubyDialog" />
         <menuitem name="ReloadRuby" action="ReloadRuby" />
--->
+
       </menu>
     </menu>
 
@@ -209,10 +209,10 @@
       <menuitem name="SaveLink" action="SaveLink"/>
       <separator name="link-pop-sep2"/>
       <menuitem name="ClipSelection" action="ClipSelection"/>
-<!--
+
       <separator name="link-pop-sep3"/>
       <menuitem name="InstallAsRubyExtension" action="InstallAsRubyExtension"/>
--->
+
     </popup>
 
     <popup name="ImagePopup">
--- origsrc/kazehakase-0.5.4/module/embed/webkit-gtk/Makefile.am	2008-03-29 03:20:40.000000000 -0500
+++ src/kazehakase-0.5.4/module/embed/webkit-gtk/Makefile.am	2008-06-02 22:35:39.421875000 -0500
@@ -31,4 +31,5 @@
 	kz-webkit-gtk.h
 
 webkit_gtk_la_LIBADD = 			\
+	$(top_builddir)/src/libkazehakase.la	\
 	$(WEBKITGTK_LIBS)
--- origsrc/kazehakase-0.5.4/src/Makefile.am	2008-03-29 03:20:42.000000000 -0500
+++ src/kazehakase-0.5.4/src/Makefile.am	2008-06-02 22:55:48.718750000 -0500
@@ -4,8 +4,7 @@
 	  widget prefs_ui sidebar libegg
 
 
-libkazehakasedir = $(libdir)/$(PACKAGE)
-libkazehakase_LTLIBRARIES = libkazehakase.la
+lib_LTLIBRARIES = libkazehakase.la
 
 AM_CPPFLAGS =						\
 	-DLOCALEDIR=\""$(localedir)"\"			\
--- origsrc/kazehakase-0.5.4/src/net/gnet.h	2008-03-29 03:20:41.000000000 -0500
+++ src/kazehakase-0.5.4/src/net/gnet.h	2008-06-02 21:57:55.265625000 -0500
@@ -28,10 +28,6 @@
 #ifndef _GNET_H
 #define _GNET_H
 
-#ifdef __CYGWIN__
-#  define __USE_W32_SOCKETS
-#endif
-
 #include <glib.h>
 #ifdef G_OS_WIN32
 #  define GNET_WIN32
--- origsrc/kazehakase-0.5.4/src/net/inetaddr.c	2008-03-29 03:20:41.000000000 -0500
+++ src/kazehakase-0.5.4/src/net/inetaddr.c	2008-06-02 21:49:04.296875000 -0500
@@ -1027,7 +1027,11 @@
 	      /* Save the address */
 	      ia = g_new0(GInetAddr, 1);
 	      ia->ref_count = 1;
+#if HAVE_INET6
 	      ia->sa.ss_family = (size == 4)? AF_INET : AF_INET6;
+#else
+	      ia->sa.ss_family = AF_INET;
+#endif
 	      GNET_INETADDR_SET_SS_LEN(ia);
 	      memcpy(GNET_INETADDR_ADDRP(ia), buf, size);
 	      GNET_INETADDR_PORT(ia) = g_htons(state->port);
